### PR TITLE
1716 rename lb stats files to data files

### LINKED ISSUE #1716: Rename LB "stats" files to "data" files
**What Needs to be Done?**

The current files that `--vt_lb_stats` output are not statistics but data files. Rename the command-line arguments and functions in VT accordingly to match proper terminology. ~We probably want to preserve the old command-line argument names for now, so we don't break scripts right away.~

### PR DESCRIPTION
**Pull request by stmcgovern**
Fixes #1716. ~~No changes to file names or command line flags.~~ (No longer true).

### CODE REVIEW

**Comment by nlslatt** [comment]
@stmcgovern The failed test writes a file and then reads it. Just looking over the code, it seems like the file write may not have been committed when the read started (despite the barrier). I remember @lifflander mentioning having bad experiences with writing to files in CI, so I'm re-running the macosx test to see if it passes this time.

File: src/vt/configs/arguments/args.cc [hunk 1]
```diff
@@ -490,12 +490,12 @@ void addLbArgs(CLI::App& app, AppConfig& appConfig) {
...
-  auto wx = app.add_option("--vt_lb_stats_dir",     appConfig.vt_lb_stats_dir,   lb_stats_dir, lbd);
-  auto wy = app.add_option("--vt_lb_stats_file",    appConfig.vt_lb_stats_file,  lb_stats_file,lbs);
-  auto xx = app.add_option("--vt_lb_stats_dir_in",  appConfig.vt_lb_stats_dir_in,  lb_stats_dir_in, lbd);
-  auto xy = app.add_option("--vt_lb_stats_file_in", appConfig.vt_lb_stats_file_in, lb_stats_file_in,lbs);
+  auto ww = app.add_flag("--vt_lb_stats",           appConfig.vt_lb_data,       lb_data);
+  auto xz = app.add_flag("--vt_lb_stats_compress",  appConfig.vt_lb_data_compress, lb_data_comp);
+  auto wx = app.add_option("--vt_lb_stats_dir",     appConfig.vt_lb_data_dir,   lb_data_dir, lbd);
+  auto wy = app.add_option("--vt_lb_stats_file",    appConfig.vt_lb_data_file,  lb_data_file,lbs);
+  auto xx = app.add_option("--vt_lb_stats_dir_in",  appConfig.vt_lb_data_dir_in,  lb_data_dir_in, lbd);
+  auto xy = app.add_option("--vt_lb_stats_file_in", appConfig.vt_lb_data_file_in, lb_data_file_in,lbs);
```
Comment (jstrzebonski) [inline src/vt/configs/arguments/args.cc:498]: Do we want to add additional options matching the new terminology (apart from leaving old ones for backward compatibility with scripts, as mentioned in the issue description)?

File: docs/md/node-stats.md [hunk 2]
```diff
@@ -21,16 +21,16 @@ can dump these to files in a VOM file (Virtual Object Map). As indicated by the
 name, the VOM file specifies the mapping of object to node for each phase along
 with statistics for each object (computation time and communication load).
```
Comment (nlslatt) [inline docs/md/node-stats.md:22]: This should say data, not statistics. Also, the filename still contains `stats`.

File: src/vt/vrt/collection/balance/node_stats.cc [hunk 3]
```diff
@@ -129,22 +129,22 @@ void NodeStats::initialize() {
...
 
 void NodeStats::createStatsFile() {
-  auto const file_name = theConfig()->getLBStatsFileOut();
-  auto const compress = theConfig()->vt_lb_stats_compress;
+  auto const file_name = theConfig()->getLBDataFileOut();
+  auto const compress = theConfig()->vt_lb_data_compress;
 
   vt_debug_print(
     normal, lb,
     "NodeStats::createStatsFile: file={}\n", file_name
```
Comment (nlslatt) [inline src/vt/vrt/collection/balance/node_stats.cc:144]: The class name, filename, and other names involving `Stats` are no longer appropriate.

File: src/vt/vrt/collection/balance/stats_restart_reader.cc [hunk 4]
```diff
@@ -72,7 +72,7 @@ void StatsRestartReader::setProxy(
 }
 
 void StatsRestartReader::startup() {
```
Comment (nlslatt) [inline src/vt/vrt/collection/balance/stats_restart_reader.cc:74]: The class name and filename are no longer appropriate.

**Comment by stmcgovern** [comment]
> This is a good start, but the changes are not extensive enough

@nlslatt Thanks for clarifying. I took a conservative approach, and I will now expand based on your comments.

File: src/vt/runnable/make_runnable.h [hunk 5]
```diff
@@ -161,53 +161,53 @@ struct RunnableMaker {
...
   template <typename ElmT>
-  RunnableMaker&& withLBStatsVoidMsg(ElmT* elm) {
-    return withLBStats(&elm->getStats(), elm->getElmID());
+  RunnableMaker&& withLBDataVoidMsg(ElmT* elm) {
+    return withLBData(&elm->getLBData(), elm->getElmID());
   }
 
   /**
-   * \brief Add LB stats for instrumentation directly with element ID and stats
+   * \brief Add LB data for instrumentation directly with element ID and stats
```
Comment (nlslatt) [inline src/vt/runnable/make_runnable.h:189]: The second instance of `stats` here also refers to `LB data`.

File: src/vt/runnable/make_runnable.h [hunk 6]
```diff
@@ -161,53 +161,53 @@ struct RunnableMaker {
...
    *
    * \param[in] stats the stats
    * \param[in] elm_id the element ID
    */
   template <typename StatsT, typename T>
-  RunnableMaker&& withLBStats(StatsT* stats, T elm_id) {
+  RunnableMaker&& withLBData(StatsT* stats, T elm_id) {
 #if vt_check_enabled(lblite)
-    impl_->template addContext<ctx::LBStats>(stats, elm_id);
+    impl_->template addContext<ctx::LBData>(stats, elm_id);
```
Comment (nlslatt) [inline src/vt/runnable/make_runnable.h:197]: `stats` and `StatsT` here both refer to LB data and so should be renamed to something more appropriate.

**Review by nlslatt — CHANGES_REQUESTED** [review]
I haven't reviewed yet for anything that got changed in error, but I wanted to start with the small number of things that got missed. There are two review comments plus what's below.

I think `\ref proc-stats` in `docs/md/collection.md` should be pointing to `node-lb-data` instead.

Also, we need to rename `StatsMapLB` (and the associated files) to something more appropriate. I've started a discussion here: https://darmatalk.slack.com/archives/C01SEK9R3M1/p1650563842991169

File: src/vt/runnable/make_runnable.h (hunk 6 above)
Comment (stmcgovern) [reply src/vt/runnable/make_runnable.h:197]: Thank you. This file was a little unclear to me.

File: src/vt/vrt/collection/balance/lb_type.h [hunk 7]
```diff
@@ -58,7 +58,7 @@ enum struct LBType : int8_t {
   , HierarchicalLB   = 2
   , RotateLB         = 3
   , TemperedLB       = 4
-  , StatsMapLB       = 5
+  , OfflineLB       = 5
```
Comment (nlslatt) [inline src/vt/vrt/collection/balance/lb_type.h:61]: Please maintain alignment.

**Review by nlslatt — COMMENTED** [review]
Amost there!
`baselb.cc` line 97: `load stats size` --> `load data size`.
`temperedlb.cc` line 229: `from vt stats` --> `from lb data`.

**Comment by nlslatt** [comment]
> Amost there! `baselb.cc` line 97: `load stats size` --> `load data size`. `temperedlb.cc` line 229: `from vt stats` --> `from lb data`.

I made these changes myself so we can get this merged before we get more merge conflicts.

**Review by nlslatt — APPROVED** [review]

**Pull request merged**